src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -477,60 +476,32 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
                 result->second = llvm::BasicBlock::Create(builder.getContext(), "", function);
             }
         };
-        switch (opCode)
-        {
-            case OpCodes::Goto:
-            {
-                auto target = consume<std::int16_t>(current);
-                addBasicBlock(target + offset);
-                break;
-            }
-            case OpCodes::GotoW:
-            {
-                auto target = consume<std::int32_t>(current);
-                addBasicBlock(target + offset);
-                break;
-            }
-            case OpCodes::IfACmpEq:
-            case OpCodes::IfACmpNe:
-            case OpCodes::IfICmpEq:
-            case OpCodes::IfICmpNe:
-            case OpCodes::IfICmpLt:
-            case OpCodes::IfICmpGe:
-            case OpCodes::IfICmpGt:
-            case OpCodes::IfICmpLe:
-            case OpCodes::IfEq:
-            case OpCodes::IfNe:
-            case OpCodes::IfLt:
-            case OpCodes::IfGe:
-            case OpCodes::IfGt:
-            case OpCodes::IfLe:
-            case OpCodes::IfNonNull:
-            case OpCodes::IfNull:
-            {
-                auto target = consume<std::int16_t>(current);
-                addBasicBlock(target + offset);
-                addBasicBlock(current.data() - code.getCode().data());
-                break;
-            }
-            default: break;
-        }
+        match(
+            operation, [&](OneOf<Goto, GotoW> gotoOp) { addBasicBlock(gotoOp.target + gotoOp.offset); },
+            [&](OneOf<IfACmpEq, IfACmpNe, IfICmpEq, IfICmpNe, IfICmpLt, IfICmpGe, IfICmpGt, IfICmpLe, IfEq, IfNe, IfLt,
+                      IfGe, IfGt, IfLe, IfNonNull, IfNull>
+                    cmpOp)
+            {
+                addBasicBlock(cmpOp.target + cmpOp.offset);
+                addBasicBlock(cmpOp.offset + sizeof(OpCodes) + sizeof(int16_t));
+            },
+            [](...) {}); // NOLINT(cert-dcl50-cpp) we need this to be the lowest priority
```

Comment:
Lets just remove this from our `.clang-tidy` file, we might need this more often and that lint in particular is not very useful to us